suite/tests/client-interface/drx_timeout_scale-test.cpp

```diff
@@ -0,0 +1,262 @@
+/* **********************************************************
+ * Copyright (c) 2025 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include <assert.h>
+#include <linux/futex.h>
+#include <math.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syscall.h>
+
+#include <atomic>
+#include <cstdint>
+#include <iostream>
+#include <thread>
+
+// This is set globally in CMake for other tests so easier to undef here.
+#undef DR_REG_ENUM_COMPATIBILITY
+
+#include "configure.h"
+#include "dr_api.h"
+#include "drx.h"
+#include "tools.h"
+
+#ifndef LINUX
+#    error Only Linux supported for this test.
+#endif
+
+namespace dynamorio {
+namespace drmemtrace {
+
+#define VERBOSE 1
+#if VERBOSE
+#    define VPRINT(...) print(__VA_ARGS__)
+#else
+#    define VPRINT(...) /* nothing */
+#endif
+
+#ifndef X64
+typedef struct _timespec64_t {
+    int64 tv_sec;
+    int64 tv_nsec;
+} timespec64_t;
+#endif
+
+static pthread_cond_t condvar;
+static bool child_ready;
+static pthread_mutex_t lock;
+static std::atomic<bool> child_should_exit;
+
+bool
+my_setenv(const char *var, const char *value)
+{
+    return setenv(var, value, 1 /*override*/) == 0;
+}
+
+static void *
+thread_routine(void *arg)
+{
+    int64_t futex_count = 0;
+
+    pthread_mutex_lock(&lock);
```

Comment:
I guess it's not enough or very easy to cleanly share test setup with suite/tests/client-interface/drx_time_scale-test.c?

Reply:
Right, I considered it briefly.

I added this top-level comment:

```
/* Tests system calls with timeouts.  This does share some boilerplate
 * with drx_sleep_scale-test.cpp and drx_time_scale-test.cpp but it's
 * not trivial to share and the other tests are too long in duration to
 * combine into one test.
 */
```